packages/url-loader/src/types/video.ts

```diff
@@ -1,13 +1,15 @@
-import { z } from 'zod';
+import { z } from "zod";
+import { assetOptionsSchema } from "./asset.js";
+import { abrProps } from "../plugins/abr.js";
 
-import { assetOptionsSchema } from './asset';
+export const videoOptionsSchema = assetOptionsSchema.merge(
+  z.object({
+    // Spreading plugins instead of extend or merge to avoid excessive schema warning
+    // https://github.com/microsoft/TypeScript/issues/34933#issuecomment-1772787785
+    ...abrProps,
+  }),
+);
 
-import { props as abrPluginProps } from '../plugins/abr';
+type _VideoOptions = z.infer<typeof videoOptionsSchema>;
 
-export const videoOptionsSchema = assetOptionsSchema.merge(z.object({
-  // Spreading plugins instead of extend or merge to avoid excessive schema warning
-  // https://github.com/microsoft/TypeScript/issues/34933#issuecomment-1772787785
-  ...abrPluginProps,
-}))
-
-export type VideoOptions = z.infer<typeof videoOptionsSchema>;
\ No newline at end of file
+export interface VideoOptions extends _VideoOptions {}
```

Comment:
question: I am not sure if this change will actually fix the issue on the vue projects. I think this would need to be tested by me. Could you let me know once you will release a `rc` version or beta of this new approach so that I can test it out and let you know if it makes a difference?

Reply:
Because eventually it is `type _VideoOptions = z.infer<typeof videoOptionsSchema>;`

Reply:
Yes, this was not the change designed to address the resolution issue with `zod`.

Take a look at `packages/url-loader/build.ts`. I only applied the strategy to `ConstructUrlProps` the primary parameter of `constructCloudinaryUrl`. However, this pattern could easily be extended to compile any other zod-inferred types down to standard TS.

Reply:
I see, thanks for clarification :)